pkg/vulnerabilityreport/scanner.go

```diff
@@ -115,12 +116,25 @@ func (s *Scanner) getCredentials(ctx context.Context, spec corev1.PodSpec, ns st
 	return kube.MapContainerNamesToDockerAuths(kube.GetContainerImagesFromPodSpec(spec), imagePullSecrets)
 }
 
-func (s *Scanner) prepareScanJob(workload kube.Object, spec corev1.PodSpec, credentials map[string]docker.Auth) (*batchv1.Job, []*corev1.Secret, error) {
+func getVulnerabilityScannerPodTolerations(starboardConfigData starboard.ConfigData) []corev1.Toleration {
+	vulnerabilityScannerPodTolerations := []corev1.Toleration{}
+	tolerationsJsonFromConfigMap := starboardConfigData["vulnerabilityScanner.podTolerations"]
+	klog.V(3).Infof("vulnerabilityScanner.podTolerations: %s", tolerationsJsonFromConfigMap)
+	err := json.Unmarshal([]byte(tolerationsJsonFromConfigMap), &vulnerabilityScannerPodTolerations)
+	if err != nil {
+		klog.V(3).Infof("error unmarshalling vulnerabilityScanner.podTolerations: %s", err)
```

Comment:
Even if the param is misconfigured (e.g. invalid JSON) would you proceed with scheduling a scan job anyway without tolerations? I think the end user won't see this Infof message unless he enabled verbose logs with the -v flag.

Reply:
I was thinking that we would.  At least that way the user will see the pods Pending and the job incomplete and konw that they need to investigate.  Would it be better to just exit from the CLI?  What about the operator?  What should it do instead?  Still log the message but not try to schedule the job?   Perhaps I should remove the V(3)?

Reply:
With the latest refactoring, it will not schedule the scan if the tolerations are misconfigured.